aten/src/ATen/native/Collectives.cpp

```diff
@@ -0,0 +1,23 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+
+#include <ATen/core/Tensor.h>
+#include <ATen/Parallel.h>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#include <ATen/NativeFunctions.h>
+#else
+#endif
+
+namespace at {
+namespace native {
+
+// Can we skip having a c++ impl? i will try doing a python_dispatcher impl
```

Comment:
I think so but we have to impede the codegen from producing the default composite entry. Maybe a explicit but empty dispatch: entry will work?